minedd/embeddings.py

```diff
@@ -0,0 +1,169 @@
+import os
+import pandas as pd
+from difflib import get_close_matches
+from tqdm import tqdm
+from paperqa import Docs
+import pickle as pkl
+from minedd.utils import configure_settings, safely_load_pickle_file
+
+
+class Embeddings:
+    """Class to process PDF papers and create or update embeddings based on them.
+    This class has an internal docs object that is used to store a list of PaperQA Doc objects.
+    Each Doc object contains internally the embeddings of chunks of a single paper plus metadata with provenance (n_docs == n_processed_papers).
+
+    Attributes:
+        model (str): Generative Model that later will be used for answering queries.
+        embedding_model (str): Encoder Model to use for creating the embeddings.
+        output_embeddings_path (str): Path to save the output embeddings file.
+        paper_directory (str): Directory containing the papers to process.
+        docs (Docs, optional): Existing Docs object, to be able to manipulate it at Runtime.
+    """
+    def __init__(self,
+                 model: str = "ollama/llama3.2:1b",
+                 embedding_model: str = "ollama/mxbai-embed-large:latest",
+                 output_embeddings_path: str = "out/embeddings.pkl",
+                 paper_directory: str = "data/",
+                 existing_docs: Docs = None,
+                 ):
+        self.output_embeddings_path = output_embeddings_path
+        self.paper_directory = paper_directory
+        self.embedding_model = embedding_model
+        self.docs = existing_docs
+
+        self.settings = configure_settings(model, self.embedding_model, self.paper_directory)
+
+        emb_directory = os.path.dirname(self.output_embeddings_path)
+        if emb_directory:
+            os.makedirs(emb_directory, exist_ok=True)
+
+
+    def __repr__(self):
```

Comment:
oh that's nice! I wasn't aware of this method __repr__ to costumise return statements/prints. Sounds very useful, I'll make sure to use it in the future